Cataclysm/Rogue.lua

```diff
@@ -515,15 +544,226 @@ spec:RegisterAuras( {
         aliasType = "debuff",
         aliasMode = "longest"
     }
-} )
+})
+
+
+spec:RegisterStateExpr( "pmultiplier", function ()
+    if not this_action then return 0 end
+
+    local a = class.abilities[ this_action ]
+    if not a then return 0 end
+
+    local aura = a.aura or this_action
+    if not aura then return 0 end
+
+    if debuff[ aura ] and debuff[ aura ].up then
+        return debuff[ aura ].pmultiplier or 1
+    end
+
+    return 0
+end )
+
+-- Bleed Modifiers
+
+local function NewBleed( key, spellID )
+    tracked_bleeds[ key ] = {
+        id = spellID,
+        rate = {},
+        last_tick = {},
+        haste = {}
+    }
+
+    tracked_bleeds[ spellID ] = tracked_bleeds[ key ]
+end
+
+local function ApplyBleed( key, target )
+    local bleed = tracked_bleeds[ key ]
+    bleed.haste[ target ]        = 100 + GetHaste()
```

Comment:
Don't use raw API calls since we could theoretically change haste values in a set of recommendations.  Use `state.stat.haste` or whatever is appropriate unless I'm overlooking something.